Tests/WebApi.UnitTests/Applications/DirectorOperations/Commands/DeleteDirector/DeleteDirectorCommandValidatorTest.cs

```diff
@@ -0,0 +1,47 @@
+using FluentAssertions;
+using MovieStoreWebApi.Applications.DirectorOperations.Commands.DeleteDirector;
+using MovieStoreWebApi.DbOperations;
+using MovieStoreWebApi.Entities;
+using Tests.WebApi.UnitTests.TestSetup;
+using Xunit;
+
+namespace Tests.WebApi.UnitTests.Applications.DirectorOperations.Commands.DeleteDirector
+{
+    public class DeleteDirectorCommandValidatorTest : IClassFixture<CommonTestFixture>
+    {
+        private readonly MovieStoreDbContext _context;
+        public DeleteDirectorCommandValidatorTest(CommonTestFixture testFixture)
+        {
+            _context = testFixture.Context;
+        }
+
+        [InlineData(0)]
+        [InlineData(-1)]
+        [InlineData(-2)]
+
+        [Theory]
+        public void WhenInvalidInputAreGiven_Validator_ShouldBeReturnErrors(int DirectorId)
+        {
+            DeleteDirectorCommand command = new DeleteDirectorCommand(null);
+            command.DirectorId = DirectorId;
+
+            DeleteDirectorCommandValidator validator = new DeleteDirectorCommandValidator();
+            var result = validator.Validate(command);
+
+            result.Errors.Count.Should().BeGreaterThan(0);
+        }
+
+        [Fact]
+        public void WhenValidInputsAreGiven_Director_ShouldBeDeleted()
+        {
+            DeleteDirectorCommand command = new DeleteDirectorCommand(_context);
+            command.DirectorId = 2;
+
+            DeleteDirectorCommandValidator validator = new DeleteDirectorCommandValidator();
+            var result = validator.Validate(command);
+
+            FluentActions.Invoking(() => command.Handle()).Invoke();
+            result.Errors.Count.Should().Be(0);
```

Comment:
Buralarda siralama bana mantiksiz geliyor. Validator'daki error sayisini daha once kontrol etmen gerekmez mi? Validator'da hata varsa command'i calistirmak istemezsin diye dusunuyorum.

Reply:
Bu testin Fact bir test olmasından dolayı command'in de Handle edilip, test sonucunda error şartına uygun olup olmadığı denetlenmelidir.

Reply:
Sordugum soru siralamayla ilgili, testin tanimi veya hatanin kontrol edilmesiyle ilgili degil.

41'de command'i validator ile validate ediyorsun, ancak hata olup olmadigina bakmadan 43'te command'i calistiriyorsun. Validator error vermisse zaten her turlu test'i fail edecek, cunku sonda kullanmissin, ancak input hataliysa ve test zaten fail olacaksa, command'i calistirmaya devam etmene gerek yok.

Bu dedigim ayrica mantiken de daha dogru bir siralama. Once bir seyi validate etmissen icerisinde hata olup olmadigina baktiktan sonra diger asamalara gecersin, yoksa validate etmenin hicbir amaci kalmaz. Uzerine hatali veriyi kullanarak islem yaparak guvenlik acigi olusturursun ancak.